pkg/collector/python/init.go

```diff
@@ -317,6 +318,9 @@ func resolvePythonExecPath(ignoreErrors bool) (string, error) {
 		log.Debugf("Executable folder is %v", _here)
 
 		embeddedPythonHome3 := filepath.Join(_here, "..", "embedded3")
+		if runtime.GOOS == "darwin" {
+			embeddedPythonHome3 = filepath.Join(_here, "../..", "embedded")
+		}
```

Comment:
Might be worth making the alternative values here explicit instead of overwriting _if darwing_, e.g.
```suggestion
		var embeddedPythonHome3 string
		if runtime.GOOS == "windows" {
			embeddedPythonHome3 = filepath.Join(_here, "..", "embedded3")
		} else { // IIRC both macOS and Linux would have the same relative paths so it makes sense as the "default" value
			embeddedPythonHome3 = filepath.Join(_here, "../..", "embedded")
		}
```